File: server/src/agents/baseAgent.js
// Shared contract for all agents. Each agent takes a task description + shared
// context and returns a structured result the orchestrator (future PlannerAgent)
// can chain into the next step.
export class BaseAgent {
  constructor(name) {
    if (this.constructor === BaseAgent) {
      throw new Error('BaseAgent is abstract and cannot be instantiated directly');
    }
    this.name = name;
  }

  // Override in subclasses. Must return { success, output, error? }.
  async run(_task, _context) {
    throw new Error(`${this.name}.run() not implemented`);
  }
}

File: server/app/agents/base_agent.py
class BaseAgent:
    """Shared contract for all agents. Each agent takes a task description +
    shared context and returns a structured result the orchestrator can chain
    into the next step."""

    def __init__(self, name: str):
        if type(self) is BaseAgent:
            raise TypeError("BaseAgent is abstract and cannot be instantiated directly")
        self.name = name

    async def run(self, task, context: dict | None = None) -> dict:
        raise NotImplementedError(f"{self.name}.run() not implemented")
